vm/src/stdlib/itertools.rs

```diff
@@ -1264,7 +1266,42 @@ mod decl {
     }
 
     #[pyimpl(with(IterNext, Constructor))]
-    impl PyItertoolsCombinations {}
+    impl PyItertoolsCombinations {
+        #[pymethod(magic)]
+        fn reduce(zelf: PyRef<Self>, vm: &VirtualMachine) -> PyTupleRef {
+            if zelf.result.read().is_none() {
+                return vm.new_tuple((
+                    zelf.class().clone(),
+                    vm.new_tuple((
+                        vm.new_tuple(zelf.pool.clone()),
+                        vm.ctx.new_int(zelf.r.load()),
+                    )),
+                ));
+            } else if zelf.exhausted.load() {
+                return vm.new_tuple((
+                    zelf.class().clone(),
+                    vm.new_tuple((vm.new_tuple(()), vm.ctx.new_int(zelf.r.load()))),
+                ));
+            } else {
+                return vm.new_tuple((
+                    zelf.class().clone(),
+                    vm.new_tuple((
+                        vm.new_tuple(zelf.pool.clone()),
+                        vm.ctx.new_int(zelf.r.load()),
+                    )),
+                    vm.ctx.new_tuple(
+                        zelf.result
+                            .read()
+                            .as_ref()
+                            .unwrap()
+                            .iter()
+                            .map(|&i| zelf.pool[i].clone())
+                            .collect(),
+                    ),
+                ));
+            }
```

Comment:
```suggestion
            if zelf.result.read().is_none() {
                vm.new_tuple((
                    zelf.class().clone(),
                    vm.new_tuple((
                        vm.new_tuple(zelf.pool.clone()),
                        vm.ctx.new_int(zelf.r.load()),
                    )),
                ))
            } else if zelf.exhausted.load() {
                vm.new_tuple((
                    zelf.class().clone(),
                    vm.new_tuple((vm.new_tuple(()), vm.ctx.new_int(zelf.r.load()))),
                ))
            } else {
                vm.new_tuple((
                    zelf.class().clone(),
                    vm.new_tuple((
                        vm.new_tuple(zelf.pool.clone()),
                        vm.ctx.new_int(zelf.r.load()),
                    )),
                    vm.ctx.new_tuple(
                        zelf.result
                            .read()
                            .as_ref()
                            .unwrap()
                            .iter()
                            .map(|&i| zelf.pool[i].clone())
                            .collect(),
                    ),
                ))
            }
```

Rust programmers prefer to not to use `return` for last expression of the function block